entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultEntitlementApi.java

```diff
@@ -245,6 +258,57 @@ public Entitlement doCall(final EntitlementApi entitlementApi, final Entitlement
     }
 
     @Override
+    public List<Entitlement> createBaseEntitlementsWithAddOns(final UUID accountId, final Iterable<BaseEntitlementWithAddOnsSpecifier> baseEntitlementSpecifiersWithAddOns, final Iterable<PluginProperty> properties, final CallContext callContext) throws EntitlementApiException {
+
+        logCreateEntitlementsWithAOs(log, baseEntitlementSpecifiersWithAddOns);
+
+        final EntitlementContext pluginContext = new DefaultEntitlementContext(OperationType.CREATE_SHOPPING_CART_SUBSCRIPTIONS,
+                                                                               accountId,
+                                                                               null,
+                                                                               baseEntitlementSpecifiersWithAddOns,
+                                                                               null,
+                                                                               properties,
+                                                                               callContext);
+
+        final WithEntitlementPlugin<List<Entitlement>> createBaseEntitlementsWithAddOns = new WithEntitlementPlugin<List<Entitlement>>() {
+            @Override
+            public List<Entitlement> doCall(final EntitlementApi entitlementApi, final EntitlementContext updatedPluginContext) throws EntitlementApiException {
+                final InternalCallContext contextWithValidAccountRecordId = internalCallContextFactory.createInternalCallContext(accountId, callContext);
+
+                try {
+                    final List<SubscriptionBase> subscriptionsWithAddOns = subscriptionBaseInternalApi.createBaseSubscriptionsWithAddOns(accountId, baseEntitlementSpecifiersWithAddOns, contextWithValidAccountRecordId);
+
+                    final DateTime entitlementRequestedDate = dateHelper.fromLocalDateAndReferenceTime(
```

Comment:
@pierre 
I understand that the effectiveDate is the correct one. The thing is that if you see [this line](https://github.com/javier-gomez-github/killbill/blob/d4ce9b37a52cd8000a5a6ca1ffa321f777ac1e5b/entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultEntitlementApi.java#L282), I'm getting the effectiveDate from the `updatedPluginContext`, not from the `SubscriptionBase` object.
If I move it to inside the `for` statement, the items I have are `SubscriptionBase` which are the recently created Subscriptions and AddOns. And basically, they don't have the `effectiveDate`, but the `createdDate`, `updatedDate`, etc.
In summary, should I iterate over the `updatedPluginContext.getBaseEntitlementWithAddOnsSpecifiers()` until I find the one that matches the `SubscriptionBase` and then use its `effectiveDate`?
I don't see it right.

Thanks!